Hunting Queries/Microsoft 365 Defender/Email Queries/Spam/Spam Detection IP and Geo Position.yaml

```diff
@@ -0,0 +1,27 @@
+id: fc2a1197-7ad5-43e3-83f1-0ecd5ada41ca
+name: Spam detection by IP and its location
+description: |
+  This query visualises total emails with Spam detections summarizing the data by email sender IP address (SenderIPv4).
+description-detailed: |
+  This query returns total emails with Spam detections summarizing the data by email sender IP address (SenderIPv4) and its geoagraphical position in Microsoft Defender for Office 365.
+  To build a graphical map please consider exporting to CSV and use tools like Power BI, Excel, or custom dashboards to plot the data on a map. 
+  Taken from the the Microsoft Defender for Office 365 Detections and Insights - Microsoft Sentinel workbook.
+  https://techcommunity.microsoft.com/blog/microsoftdefenderforoffice365blog/part-2-build-custom-email-security-reports-and-dashboards-with-workbooks-in-micr/4411303
+requiredDataConnectors:
+- connectorId: MicrosoftThreatProtection
+  dataTypes:
+  - EmailEvents
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1566
+query: |
+  //This query visualises total emails with Spam detections summarizing the data by email sender IP address (SenderIPv4).
+  EmailEvents 
+  | where ThreatTypes has "Spam" 
+  | where TimeGenerated > ago(30d) // last 30 days by default, replace 30d with the desired period
+  | summarize count() by SenderIPv4
```

Comment:
In some case, SenderIPv4 field is empty. Do you want to filter those before summarizing?